src/main/java/io/snyk/jenkins/SnykToHTML.java

```diff
@@ -3,61 +3,68 @@
 import static io.snyk.jenkins.config.SnykConstants.SNYK_TEST_REPORT_JSON;
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 public class SnykToHTML {
-  public static void generateReport(
+  public static FilePath generateReport(
     SnykContext context,
     SnykInstallation installation
-  ) throws IOException, InterruptedException {
-    TaskListener log = context.getTaskListener();
-    FilePath workspace = context.getWorkspace();
-    Launcher launcher = context.getLauncher();
-    EnvVars env = context.getEnvVars();
-
-    ArgumentListBuilder args = new ArgumentListBuilder();
-
-    FilePath snykReportJson = workspace.child(SNYK_TEST_REPORT_JSON);
-    if (!snykReportJson.exists()) {
-      log.getLogger().println("Snyk report json doesn't exist");
-      return;
-    }
+  ) {
+    try {
+      FilePath workspace = context.getWorkspace();
+      Launcher launcher = context.getLauncher();
+      EnvVars env = context.getEnvVars();
+
+      FilePath snykReportJson = workspace.child(SNYK_TEST_REPORT_JSON);
+      if (!snykReportJson.exists()) {
+        throw new RuntimeException("Snyk Report JSON does not exist.");
+      }
 
-    workspace.child(SNYK_REPORT_HTML).write("", UTF_8.name());
+      FilePath reportPath = workspace.child(getURLSafeDateTime() + "_" + SNYK_REPORT_HTML);
+      reportPath.write("", UTF_8.name());
+
+      ArgumentListBuilder args = new ArgumentListBuilder()
+        .add(installation.getReportExecutable(launcher))
+        .add("-i", SNYK_TEST_REPORT_JSON);
 
-    args.add(installation.getReportExecutable(launcher));
-    args.add("-i", SNYK_TEST_REPORT_JSON, "-o", SNYK_REPORT_HTML);
-    try {
       int exitCode = launcher.launch()
         .cmds(args)
         .envs(env)
+        .stdout(reportPath.write())
         .quiet(true)
         .pwd(workspace)
         .join();
-      boolean success = exitCode == 0;
-      if (!success) {
-        log.getLogger().println("Generating Snyk html report was not successful");
+
+      if (exitCode != 0) {
+        throw new RuntimeException("Exit Code: " + exitCode);
       }
-      String reportWithInlineCSS = workspace.child(SNYK_REPORT_HTML).readToString();
-      String finalHtmlReport = ReportConverter.getInstance().modifyHeadSection(
-        reportWithInlineCSS,
+
+      String reportContents = ReportConverter.getInstance().modifyHeadSection(
+        reportPath.readToString(),
         Jenkins.get().servletContext.getContextPath()
       );
-      workspace.child(workspace.getName() + "_" + SNYK_REPORT_HTML).write(finalHtmlReport, UTF_8.name());
-    } catch (IOException ex) {
-      Util.displayIOException(ex, log);
-      ex.printStackTrace(log.fatalError("Snyk-to-Html command execution failed"));
+
+      reportPath.write(reportContents, UTF_8.name());
+
+      return reportPath;
+    } catch (IOException | InterruptedException | RuntimeException ex) {
+      throw new RuntimeException("Failed to generate report.", ex);
```

Comment:
I wrapped the entire method in a try catch so there can be shared context for any errors coming from this.